SendGrid/Example/Program.cs

```diff
@@ -1,50 +1,107 @@
-
-		    SendAsync(myMessage);
-
-			Console.ReadLine();
-		}
-
-        private static async void SendAsync(SendGridMessage message)
-	    {
-            // Create credentials, specifying your user name and password.
-            var credentials = new NetworkCredential("username", "password");
+    internal class Program
+    {
+        private static void Main()
+        {
+            // Test sending email 
+            string to = "example@example.com";
+            string from = "example@example.com";
+            string fromName = "Jane Doe";
+            SendEmail(to, from, fromName);
+            // Test viewing, creating, modifying and deleting API keys through our v3 Web API 
+            ApiKeys();
+        }
+
+        private static void SendAsync(SendGrid.SendGridMessage message)
+        {
+            // Create credentials, specifying your user Name and password.
+            string username = Environment.GetEnvironmentVariable("SENDGRID_USERNAME");
+            string password = Environment.GetEnvironmentVariable("SENDGRID_PASSWORD");
+            //string apikey = Environment.GetEnvironmentVariable("SENDGRID_APIKEY");
+            var credentials = new NetworkCredential(username, password);
 
             // Create a Web transport for sending email.
-            var transportWeb = new Web(credentials);
+            var transportWeb = new SendGrid.Web(credentials);
+            //var transportWeb2 = new SendGrid.Web(apikey);
 
             // Send the email.
             try
             {
-                await transportWeb.DeliverAsync(message);
-                Console.WriteLine("Sent!");
+                transportWeb.DeliverAsync(message).Wait();
+                Console.WriteLine("Email sent to " + message.To.GetValue(0));
+                Console.WriteLine("Press any key to continue.");
+                Console.ReadKey();
             }
             catch (Exception ex)
             {
                 Console.WriteLine(ex.Message);
+                Console.WriteLine("Press any key to continue.");
+                Console.ReadKey();
             }
-	    }
-	}
-}
\ No newline at end of file
+        }
+
+        private static void SendEmail(string to, string from, string fromName)
+        {
+            // Create the email object first, then add the properties.
+            var myMessage = new SendGrid.SendGridMessage();
+            myMessage.AddTo(to);
+            myMessage.From = new MailAddress(from, fromName);
+            myMessage.Subject = "Testing the SendGrid Library";
+            myMessage.Text = "Hello World! %tag%";
+
+            var subs = new List<String> { "私は%type%ラーメンが大好き" };
+            myMessage.AddSubstitution("%tag%", subs);
+            myMessage.AddSection("%type%", "とんこつ");
+
+            SendAsync(myMessage);
+        }
+
+        private static void ApiKeys()
+        {
+            String apiKey = Environment.GetEnvironmentVariable("SENDGRID_APIKEY", EnvironmentVariableTarget.User);

```

Comment:
Awesome...i see you needed a little refactoring to make this work.  But that's good...because now users of this api can take advantage of async/await if they want.